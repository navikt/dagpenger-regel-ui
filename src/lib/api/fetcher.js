function handleErrors(response) {
  if (!response.ok) {
    throw Error(response.statusText);
  }
  return response;
}

export const fetcher = (uri, options) =>
  fetch(uri, options)
    .then(handleErrors)
    .then((res) => res.json());

export function audience(app) {
  return `api://${process.env.NAIS_CLUSTER_NAME}.teamdagpenger.${app}/.default`;
}
