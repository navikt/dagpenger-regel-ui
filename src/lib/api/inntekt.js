import { join } from "path";

const rootUri = "/v1/inntekt";

const uklassifisert = (aktørId, vedtakId, beregningsDato) =>
  join(
    process.env.INNTEKT_API,
    rootUri,
    "uklassifisert",
    aktørId,
    "vedtak",
    vedtakId,
    beregningsDato
  );

export const uncached = (aktørId, vedtakId, beregningsDato) =>
  join(
    process.env.INNTEKT_API,
    rootUri,
    "uklassifisert/uncached",
    aktørId,
    "vedtak",
    vedtakId,
    beregningsDato
  );

export const verdikoder = () =>
  join(process.env.INNTEKT_API, rootUri, "verdikoder");

export async function postUklassifisert(
  aktorId,
  vedtakId,
  beregningsDato,
  apiToken,
  body
) {
  const data = await fetch(uklassifisert(aktorId, vedtakId, beregningsDato), {
    method: "POST",
    headers: {
      Accept: `application/json`,
      Authorization: `Bearer ${apiToken}`,
      "Content-type": "application/json",
    },
    body,
  });

  return data.json();
}

export async function getUklassifisert(
  aktorId,
  vedtakId,
  beregningsDato,
  apiToken
) {
  const res = await fetch(uklassifisert(aktorId, vedtakId, beregningsDato), {
    headers: {
      Accept: `application/json`,
      Authorization: `Bearer ${apiToken}`,
    },
  });

  return res.json();
}

export default {
  getUklassifisert,
  postUklassifisert,
};
