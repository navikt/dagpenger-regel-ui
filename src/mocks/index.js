async function initMocks() {
  if (typeof window === "undefined") {
    const { server } = await import("./server");
    server.listen({ onUnhandledRequest: "warn" });
  } else {
    const { worker } = await import("./browser");
    worker.start({ onUnhandledRequest: "warn" });
  }
}

initMocks();

export {};
