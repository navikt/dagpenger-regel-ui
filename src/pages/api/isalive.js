export default function (req, res) {
  res.status(200).end("alive");
}
