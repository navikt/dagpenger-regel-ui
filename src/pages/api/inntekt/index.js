import inntekt from "../../../lib/api/inntekt";
import { getSession } from "@navikt/dp-auth/session";
import { provider } from "../../../middleware";

export default async function handler(req, res) {
  switch (req.method) {
    case "GET":
      await handleGet(req, res);
      break;
    case "POST":
      await handlePost(req, res);
      break;
    default:
      res.status(405).text("Method Not Allowed");
      break;
  }
}

async function handleGet(req, res) {
  const session = await getSession(provider, { req });
  if (!session) return res.status(401).end();

  const { aktorId, vedtakId, beregningsDato } = req.query;
  const apiToken = await session.apiToken(process.env.INNTEKT_API_AUDIENCE);

  const data = await inntekt.getUklassifisert(
    aktorId,
    vedtakId,
    beregningsDato,
    apiToken
  );

  res.json(data);
}

async function handlePost(req, res) {
  const session = await getSession(provider, { req });
  if (!session) return res.status(401).end();

  const { aktorId, vedtakId, beregningsDato } = req.query;
  const apiToken = await session.apiToken(process.env.INNTEKT_API_AUDIENCE);

  let response = await inntekt.postUklassifisert(
    aktorId,
    vedtakId,
    beregningsDato,
    apiToken,
    req.body
  );
  res.json(response);
}
