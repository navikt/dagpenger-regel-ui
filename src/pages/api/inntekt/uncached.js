import { uncached } from "../../../lib/api/inntekt";
import { getAzureSession, getInntektOboToken } from "../../../lib/Utils/auth";

export default async function handler(req, res) {
  switch (req.method) {
    case "GET":
      await handleGet(req, res);
      break;
    case "POST":
      await handlePost(req, res);
      break;
    default:
      res.status(405).text("Method Not Allowed");
      break;
  }
}

async function handleGet(req, res) {
  const session = await getAzureSession(request);
  if (!session) return res.status(401).end();
  const apiToken = await getInntektOboToken(session);

  const { aktorId, vedtakId, beregningsDato } = req.query;

  const data = await fetch(uncached(aktorId, vedtakId, beregningsDato), {
    headers: {
      Accept: `application/json`,
      Authorization: `Bearer ${apiToken}`,
    },
  });

  res.json(await data.json());
}

async function handlePost(req, res) {
  console.log(`Forsøker å oppdatere inntekt`);
  const session = await getAzureSession(request);
  if (!session) return res.status(401).end();
  const apiToken = await getInntektOboToken(session);

  const { aktorId, vedtakId, beregningsDato } = req.query;

  try {
    const data = await fetch(uncached(aktorId, vedtakId, beregningsDato), {
      method: "POST",
      headers: {
        Accept: `application/json`,
        Authorization: `Bearer ${apiToken}`,
        "Content-type": "application/json",
      },
      body: req.body,
    });

    const jsonData = await data.json();
    console.log(`Oppdatert inntekt OK`);
    res.json(jsonData);
  } catch (e) {
    console.log(
      `Klarte ikke å håndtere POST mot uklassifisert inntekt. Feilmelding: ${e.message}`
    );
    res.status(500).send("Klarte ikke å oppdatere inntekt");
  }
}
