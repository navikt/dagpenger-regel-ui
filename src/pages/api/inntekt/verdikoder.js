import { verdikoder } from "../../../lib/api/inntekt";

export default async function handler(req, res) {
  const data = await fetch(verdikoder());

  res.json(await data.json());
}
