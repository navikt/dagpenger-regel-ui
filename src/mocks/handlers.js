import { rest } from "msw";
import verdikoder from "./fixtures/verdikoder.json";
import inntektUklassifisert from "./fixtures/inntekt-uklassifisert.json";
import enhetsregisteret from "./fixtures/enhetsregisteret.json";

export const handlers = [
  rest.get("/api/inntekt/verdikoder", (req, res, ctx) =>
    res(ctx.json(verdikoder))
  ),
  rest.get("/api/inntekt", (req, res, ctx) =>
    res(ctx.json(inntektUklassifisert))
  ),
  rest.post("/api/inntekt", (req, res, ctx) =>
    res(ctx.json({ success: true }))
  ),
  rest.get("/api/enhet/:enhetId", (req, res, ctx) =>
    res(ctx.json(enhetsregisteret))
  ),
];
