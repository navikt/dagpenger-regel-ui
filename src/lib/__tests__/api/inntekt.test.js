import { getUklassifisert, postUklassifisert } from "../../api/inntekt";
import { server } from "../../../mocks/server";
import { rest } from "msw";

describe("Inntekt API", () => {
  it("POST /uklassifisert", async () => {
    server.use(
      rest.post(
        "http://dp-inntekt-api.test/v1/inntekt/uklassifisert/123/vedtak/123/2020-01-01",
        (req, res, ctx) => {
          if (req.headers.get("content-type") != "application/json")
            return res(ctx.status(500));
          return res(
            ctx.json({
              inntekt: { arbeidsInntektMaaned: [1, 2, 3, 4, 5, 6] },
            })
          );
        }
      )
    );
    const data = await postUklassifisert(
      "123",
      "123",
      "2020-01-01",
      "123",
      "{}"
    );

    expect(data.inntekt.arbeidsInntektMaaned).toHaveLength(6);
    expect(data).toMatchSnapshot();
  });
});
