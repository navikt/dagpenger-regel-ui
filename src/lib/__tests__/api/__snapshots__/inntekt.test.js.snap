// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inntekt API POST /uklassifisert 1`] = `
Object {
  "inntekt": Object {
    "arbeidsInntektMaaned": Array [
      1,
      2,
      3,
      4,
      5,
      6,
    ],
  },
}
`;
